src/main/java/net/minestom/server/item/ItemStack.java

```diff
@@ -1,903 +1,207 @@
-
-        if (material == Material.CROSSBOW)
-            return new CrossbowMeta();
-
-        if (material == Material.WRITABLE_BOOK)
-            return new WritableBookMeta();
-
-        if (material == Material.WRITTEN_BOOK)
-            return new WrittenBookMeta();
-
-        if (material == Material.FIREWORK_STAR)
-            return new FireworkEffectMeta();
-
-        if (material == Material.FIREWORK_ROCKET)
-            return new FireworkMeta();
-
-        if (material == Material.PLAYER_HEAD)
-            return new PlayerHeadMeta();
-
-        if (material == Material.LEATHER_HELMET ||
-                material == Material.LEATHER_CHESTPLATE ||
-                material == Material.LEATHER_LEGGINGS ||
-                material == Material.LEATHER_BOOTS)
-            return new LeatherArmorMeta();
-
-        return null;
+    @Contract(pure = true)
+    public @NotNull ItemStore getStore() {
+        return store;
     }
 
-    /**
-     * Creates a {@link NBTCompound} containing the data of this item.
-     * <p>
-     * WARNING: modifying the returned nbt will not affect the item.
-     *
-     * @return this item nbt
-     */
-    @NotNull
-    public NBTCompound toNBT() {
-        NBTCompound compound = new NBTCompound()
-                .setByte("Count", amount)
-                .setString("id", material.getName());
-        if (hasNbtTag()) {
-            NBTCompound additionalTag = new NBTCompound();
-            NBTUtils.saveDataIntoNBT(this, additionalTag);
-            compound.set("tag", additionalTag);
-        }
-        return compound;
+    @Contract(pure = true)
+    public boolean isAir() {
+        return material.equals(Material.AIR);
     }
 
-    /**
-     * WARNING: not implemented yet.
-     * <p>
-     * This is be called each time an item is serialized to be send to a player,
-     * can be used to customize the display of the item based on player data.
-     *
-     * @param player the player
-     * @return the custom {@link ItemDisplay} for {@code player},
-     * null to use the normal item display name &amp; lore
-     */
-    public ItemDisplay getCustomDisplay(Player player) {
-        throw new UnsupportedOperationException("Not implemented yet");
+    @Contract(pure = true)
+    public boolean isSimilar(@NotNull ItemStack itemStack) {
+        return material.equals(itemStack.material) &&
+                meta.equals(itemStack.meta);
     }
 
     @Override
-    public @NotNull HoverEvent<ShowItem> asHoverEvent(@NotNull UnaryOperator<ShowItem> op) {
-        return HoverEvent.showItem(op.apply(ShowItem.of(this.material, this.amount, NBTUtils.asBinaryTagHolder(this.toNBT().getCompound("tag")))));
-    }
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
 
-    // Callback events
+        ItemStack itemStack = (ItemStack) o;
+        if (uuid.equals(itemStack.uuid)) return true;
```

Comment:
I think we should do more than compare the UUID. Would two items created that are the same not have different UUIDs?

Reply:
Honestly we could just remove the identifier, as we do not plan to introduce server-side data it is currently unused